src/tests/test_unittest.py

```diff
@@ -0,0 +1,44 @@
+""" tests/test_unittest.py
+This module does unit testing on main functionalitiesusing standard library unittest.
+"""
+import unittest
+from unittest.mock import patch
+
+from agents.concrete_agent import ConcreteAgent
+
+
+class TestConcreteAgent(unittest.IsolatedAsyncioTestCase):
+    def setUp(self):
+        self.agent = ConcreteAgent()
+
+    async def test_handle_custom_message(self):
+        """
+        This method tests the handling of incoming messages.
+        """
+        message_with_hello = {"type": "custom", "content": "hello world"}
+        message_without_hello = {"type": "custom", "content": "random message"}
+
+        # Testing: message with "hello"
+        with patch("agents.concrete_agent.logging.info") as mock_logging_info:
+            await self.agent.handle_custom_message(message_with_hello)
+            mock_logging_info.assert_called_once_with(f"Received message: {message_with_hello}")
+
+        # Testing: message without "hello"
+        with patch("agents.concrete_agent.logging.info") as mock_logging_info:
+            await self.agent.handle_custom_message(message_without_hello)
+            mock_logging_info.assert_not_called()
```

Comment:
Could you rewrite this to avoid code duplication and parametrize the unit test instead?

Reply:
Parameterization in test cases has been added and duplication is removed and it can bee seen in the new commit.